### PR TITLE
Problem: no attested decryption query protocol (CRO-274)

### PR DESCRIPTION
**Pull request by tomtau**
Solution: added a limited prototype TDQE server that includes the attested report in TLS cert

### CODE REVIEW

File: tx-query/app/src/enclave_u/mod.rs [hunk 1]
```diff
@@ -1,19 +1,229 @@
...
+        .path(storage_path())
+        .read_only(true);
+
+    let db = Db::start(config.build()).expect("failed to open a storage path");
+    db.open_tree(TX_KEYSPACE)
+        .expect("failed to open a tx keyspace")
+}
+
+thread_local! {
+    static TXDB: Arc<Tree> = init_tx_db();
```
Comment (devashishdxt) [inline tx-query/app/src/enclave_u/mod.rs]: Why to wrap `Tree` in an `Arc` if it is `thread_local`?

File: tx-query/app/src/main.rs [hunk 2]
```diff
@@ -26,6 +35,35 @@ fn main() {
             return;
         }
     };
-    // FIXME: it's just a dummy now
-    assert!(check_initchain(enclave.geteid(), 0xab).is_ok());
+
+    // needs to get a lock in enclave_u mod -- FIXME: I guess this is broken if sled is simultaneously opened by tx-validation and tx-query?
```
Comment (devashishdxt) [inline tx-query/app/src/main.rs]: I don't know the usecase here. But, `sled::Db` (at any given `storage_path`) cannot be opened simultaneously by two separate applications.

File: tx-query/app/src/enclave_u/mod.rs (hunk 1 above)
Comment (tomtau) [reply tx-query/app/src/enclave_u/mod.rs]: That's what's returned from `db.open_tree(TX_KEYSPACE)`: https://docs.rs/sled/0.25.0/sled/struct.Db.html#method.open_tree

File: tx-query/app/src/main.rs (hunk 2 above)
Comment (tomtau) [reply tx-query/app/src/main.rs]: The original planned use case was:

* tx-validation process: opens the sled db and writes sealed TX data into it, as it goes

* (optional) tx-query process: opens the sled db as read-only and uses it to serve decryption query requests

Comment (tomtau) [reply tx-query/app/src/main.rs]: but I guess it may need to be re-architected

File: tx-query/app/src/enclave_u/mod.rs (hunk 1 above)
Comment (tomtau) [reply tx-query/app/src/enclave_u/mod.rs]: But I guess it may go away https://github.com/crypto-com/chain-tx-enclave/pull/24#discussion_r315122508

**Pull request merged**